packages/terra-menu/src/MenuItem.jsx

```diff
@@ -29,22 +45,22 @@ const propTypes = {
   isDisabled: PropTypes.bool,
 
   /**
-   * Indicates if the item is selected. A selected item is indicated with a checkmark.
-   */
-  isSelected: PropTypes.bool,
+  * Indicates if the item is toggled. A toggled item is indicated with a checkmark.
+  */
+  isToggled: PropTypes.bool,
 
   /**
-   * Indicates if the item should be selectable.
-   */
-  isSelectable: PropTypes.bool,
+  * Indicates if the item should be toggleable.
+  */
+  isToggleable: PropTypes.bool,
```

Comment:
you need to re-consider this !! As change in prop name would be a non-passive change and requires Major version release.
I would suggest to keep the prop name and description same.

Reply:
we renamed the props to differentiate the difference between what toggleable is and what selectable is, the checkmark was acting as both toggleable and selectable, that was not exactly what terra-tabs needed, if we kept the checkmark as selected, then when inside a tab with an icon, you would have a conflict over what would win out, the icon or the checkmark, therefore, toggleable became the checkmark and selected became the highlighting

selectable is now controlled by a custom prop that terra-tabs uses which is represented by the highlighting of the menu-item, toggleable is still represented as the checkmark

@neilpfeiffer can provider further explanation if needed

Reply:
> selectable is now controlled by a custom prop that terra-tabs uses which is represented by the highlighting of the menu-item, toggleable is still represented as the checkmark

I think we would not need customProps for selectable since both will have same value we can use only one of them.

Since you are renaming existing props it would be non-passive change.  I am thinking if we can retain the older props and just keep the highlighting part of menu-item.

would like to know your thoughts @neilpfeiffer

Reply:
we did update the props for the scenario where menu item is selectable but not toggleable.  The problem is that the selectable props were used to display the checkmark (ie toggle it on and off) and not actually select the item (now done with the highlight).  If we introduced new props to toggle the checkmark on and off and changed selectable to highlight the row, that would be a non-passive change

it was done this way as is it passive and renames the prop names to be functionally correct in toggling the checkmark on and off and now a new hidden prop that is only used by terra-tabs, isHighlighted, to show the menuitem in a highlighted/selected state.

Reply:
both new props and old props are used in same way !! I don't see any impact in removing new props scenarios of selectable and check mark still works with existing `isSelectable` and `isSelected` and new prop `isHighlighted`. 

Updating first class props to custom props might cause some confusion for consumers when they refer to new props table : https://engineering.cerner.com/terra-ui/components/cerner-terra-framework-docs/menu/menu#menuitem-props-table

Reply:
the renaming of the props were done mainly for the UX perspective in that selectable was not meant to be adding a checkmark, toggleable is the term they want to use for that and to separate the idea of selectable vs toggleable, going forward, they wanted toggleable to be the verbage used

I understand it may cause some confusion when looking at the new props table initially, but I believe it is easy enough to understand and move to using the new prop if they look at the changelogs and documentation

Reply:
well if that is case then I would suggest you to keep the old props as well on props table with Deprecating labels on them similar to this : https://github.com/cerner/terra-core/blob/e5e9496580d87d558d18fe1f69c722bcfb5410e9/packages/terra-icon/src/IconBase.jsx#L38-L42

and also an ToDo comment for removing the code where customProps.isSelected is used on next major version release.

Reply:
update:
https://github.com/cerner/terra-framework/pull/1592/commits/161b5e5054cdb3903074b94d900dc11ea73f43a1